ibm/service/codeengine/resource_ibm_code_engine_app.go

```diff
@@ -0,0 +1,1006 @@
+	return &schema.Resource{
+		CreateContext: resourceIbmCodeEngineAppCreate,
+		ReadContext:   resourceIbmCodeEngineAppRead,
+		UpdateContext: resourceIbmCodeEngineAppUpdate,
+		DeleteContext: resourceIbmCodeEngineAppDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"project_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_app", "project_id"),
+				Description:  "The ID of the project.",
+			},
+			"image_reference": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_app", "image_reference"),
+				Description:  "The name of the image that is used for this job. The format is `REGISTRY/NAMESPACE/REPOSITORY:TAG` where `REGISTRY` and `TAG` are optional. If `REGISTRY` is not specified, the default is `docker.io`. If `TAG` is not specified, the default is `latest`. If the image reference points to a registry that requires authentication, make sure to also specify the property `image_secret`.",
+			},
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_app", "name"),
+				Description:  "The name of the app. Use a name that is unique within the project.",
+			},
+			"image_port": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     8080,
+				Description: "Optional port the app listens on. While the app will always be exposed via port `443` for end users, this port is used to connect to the port that is exposed by the container image.",
+			},
+			"image_secret": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_app", "image_secret"),
+				Description:  "Optional name of the image registry access secret. The image registry access secret is used to authenticate with a private registry when you download the container image. If the image reference points to a registry that requires authentication, the app will be created but cannot reach the ready status, until this property is provided, too.",
+			},
+			"managed_domain_mappings": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "local_public",
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_app", "managed_domain_mappings"),
+				Description:  "Optional value controlling which of the system managed domain mappings will be setup for the application. Valid values are 'local_public', 'local_private' and 'local'. Visibility can only be 'local_private' if the project supports application private visibility.",
+			},
+			"run_arguments": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
+				MinItems:    0,
```

Comment:
I see your point. Actually it is not necessary to provide the property, but the response will contain an empty in that case. Would you recommend to omit the `MinItems` and add a custom check to ignore a diff in case the user sets `null` (which obviously is not the same as an empty list)?